src/Storages/ObjectStorage/StorageObjectStorage.h

```diff
@@ -193,10 +211,17 @@ class StorageObjectStorage::Configuration
     /// buckets in S3. If object storage doesn't have any namepaces return empty string.
     virtual std::string getNamespaceType() const { return "namespace"; }
 
-    virtual Path getFullPath() const { return ""; }
-    virtual Path getPath() const = 0;
-    virtual void setPath(const Path & path) = 0;
+    // Path provided by the user in the query
+    virtual Path getRawPath() const = 0;
+    // Path used for reading, it is usually a globbed path like `'table_root/**.parquet'
+    Path getReadingPath() const;
+    // Path used for writing, it should not be globbed and might contain a partition key
+    Path getWritingPath(const std::string & partition_id = "") const;
```

Comment:
"Reading/writing path" is grammatically strange.
```suggestion
    Path getPathForRead() const;
    // Path used for writing, it should not be globbed and might contain a partition key
    Path getPathForWrite(const std::string & partition_id = "") const;
```